feat: add exhaust node resources fault

preespp opened the pull request:
Added a new fault: exhaust-node-resources and a corresponding incident spec (Incident 297) from issue #297

rohanarora commented on sre/roles/faults/tasks/inject_custom_exhaust_node_resources.yaml:
```diff
@@ -0,0 +1,81 @@
+              app: fake-resource-hog
+          spec:
+            nodeSelector:
+              itbench-fault: exhaust-node
```
Also if the observability tools are deployed to the same node the tool-related / observability stack-related pods will start to suffer. So @preespp as a part of this PR you would want to introduce labels so that the tool-related stack related pods get deployed to those nodes and then the application gets deployed to the other available nodes. And then that way the label itself points to non-tool nodes (aka anti-affinity rule) hence making it obscure.

rohanarora commented on sre/roles/incidents/files/ground_truths/incident_297.yaml:
```diff
@@ -0,0 +1,48 @@
+---
+fault:
+  - entity:
+      name: otel-cartservice
```
I would think the faulty entity here is the one hogging the resource.

rohanarora commented on sre/roles/incidents/files/ground_truths/incident_297.yaml:
```diff
@@ -0,0 +1,48 @@
+
+alerts:
+  - id: PendingPodsDetected
+    group_id: otel-cartservice
```
I would try and follow the naming convention we see at https://github.com/itbench-hub/ITBench-Scenarios/blob/main/sre/roles/incidents/files/ground_truths/incident_30.yaml

rohanarora commented on sre/roles/incidents/files/ground_truths/incident_297.yaml:
```diff
@@ -0,0 +1,48 @@
+  - id: PendingPodsDetected
+    group_id: otel-cartservice
+    metadata:
+      description: Some pods remain in Pending state due to insufficient CPU/memory on the node
```
This should not be the case. We should not be seeing any pending pods.

Red-GV commented on sre/roles/incidents/files/specs/incident_297.yaml:
```diff
@@ -0,0 +1,31 @@
+        exhaust_node_resources:
+          workload:
+            kind: Deployment
+            name: loadgenerator
```
The workload should probably be one of the microservices used.

preespp commented:
Thank you for your comment. Below are what I changed based on what we discussed in last Thursday's meeting.
- Changed node label from 'itbench-fault: exhaust-node' to 'node-role.kubernetes.io/app: "true"' 
- Introduced anti-affinity so that observability/tool-related pods are deployed to separate nodes, preventing them from being affected by the resource hog.
- Ensured the targeted workload is scheduled on a different node to avoid pending pods.
- Updated incident metadata and alerts to reflect resource pressure correctly.
- Changed tasks to label/unlabel nodes before/after resource hog deployment to maintain cluster state.

Red-GV commented on sre/roles/faults/tasks/inject_custom_exhaust_node_resources.yaml:
```diff
@@ -0,0 +1,51 @@
+        name: "{{ target_node }}"
+        labels:
+          node-role.kubernetes.io/app: "true"
+    state: present
```
```suggestion
    state: patched
```

preespp commented on sre/roles/incidents/files/specs/incident_297.yaml:
```diff
@@ -0,0 +1,27 @@
+metadata:
+  complexity: Medium
+  id: 297
```
Yes I changed it to 41. At first I was following my assigned issue's number.
